### PR TITLE
Try using helix sdk payload support instead

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

**Comment by davidfowl** [comment]
Very nice! This will mean some of the work will make it to asp.net core for free 😀

**Comment by Pilchie** [comment]
:eyes:

**Comment by davidfowl** [comment]
The only problem is that this doesn't currently support `DOTNET_LOOKUP`

**Comment by HaoK** [comment]
What does DOTNET_LOOKUP do?

**Comment by davidfowl** [comment]
> What does DOTNET_LOOKUP do?

Fixed it, DOTNET_ROOT 😄

**Comment by HaoK** [comment]
Is it a problem that we install the runtime to DOTNET_ROOT and the sdk is wherever?  Alternatively, if we change things for helix to always set DOTNET_ROOT, we can just install the runtime to that variable maybe?

**Comment by davidfowl** [comment]
> Is it a problem that we install the runtime to DOTNET_ROOT and the sdk is wherever? Alternatively, if we change things for helix to always set DOTNET_ROOT, we can just install the runtime to that variable maybe?

Yep, that's what I'm planning to do 😄

**Comment by davidfowl** [comment]
Here https://github.com/dotnet/helix-diagnostic-prototype/blob/a16adb440512c92dff61371c5a1ee7e1a43c94f7/src/overrides/Helix.Sdk/tools/dotnet-cli/DotNetCli.targets#L14-L23

**Comment by HaoK** [comment]
Oh right, I can try this in my branch in your repo first

**Comment by HaoK** [comment]
Okay some of this is starting to come back to me now...  I don't think this will work, if we try to install to the correlation payloads, it will only work on windows (and even then its frowned on), and this won't work on unix since I believe those directories end up being read only, which is why our shell scripts use a random directory in the workitem root to install sdk/runtime...

**Comment by HaoK** [comment]
I think what we need is the ability to ask for both sdk and runtime, something like
```
<IncludeDotNetSdk>5.0.100-preview.2.20120.3</IncludeDotNetSdk>
<IncludeDotNetRuntime>5.0.100-preview.4.1324</IncludeDotNetRuntime>
```
Basically move this functionality into helix/arcade

File: eng/helix/helix.proj [hunk 1]
```diff
@@ -23,6 +23,9 @@
     <SkipInvalidConfigurations>true</SkipInvalidConfigurations>
     <MaxRetryCount Condition="'$(MaxRetryCount)' == ''">2</MaxRetryCount>
     <HelixAccessToken Condition="'$(_UseHelixOpenQueues)' != 'true'">$(HelixApiAccessToken)</HelixAccessToken>
+    <IncludeDotNetCli>true</IncludeDotNetCli>
+    <DotNetCliPackageType>sdk</DotNetCliPackageType>
+    <DotNetCliVersion>$(NETCoreSdkVersion)</DotNetCliVersion>
```
Comment (rynowak) [inline eng/helix/helix.proj:28]: oooh this is nice. Is this new?

**Comment by HaoK** [comment]
Okay so I guess it looks like it just 'works', our shell scripts were able to install the runtime into the DOTNET_ROOT directory :shrug:

`DOTNET_ROOT=/home/helixbot/work/B554096C/p/dotnet`

https://helix.dot.net/api/2019-06-17/jobs/ec9c0b5c-c973-4d62-b833-2b03f02dd714/workitems/AspNetTestProject1.dll/files/console.c77b38f2.log

So I guess we just need to wait for https://github.com/dotnet/arcade/pull/5284 to make it in and down to us and we can kick this PR again

**Comment by HaoK** [comment]
Replaced by https://github.com/dotnet/aspnetcore/pull/23585

**Pull request closed without merging**